### ISSUE TITLE
Moving webworker to cockle

### ISSUE THREAD

**Issue by ianthomas23** [jupyterlite/terminal#19]
Cockle now includes the WebWorker code and handling of separate WASM and JS wrapper files (jupyterlite/cockle#38, jupyterlite/cockle#40, jupyterlite/cockle#44) and I am now trying to use that here, so I have removed the local WebWorker code in branch https://github.com/jupyterlite/terminal/tree/remove-webworker and struggling to get it all correctly working together. Loading the `shell_worker.js` is failing.

To reproduce it is necessary to use the `main` branch of cockle as the changes there have not been released yet. So get commit `41cc8b014aa8` of cockle and build it locally following the instructions it its README. If you also follow the instructions for the `demo` you'll be able to run it locally in a browser and confirm that commands like `ls` and `grep` work in the WebWorker. Then run `npm link` on cockle.

In the same micromamba environment get https://github.com/jupyterlite/terminal/tree/remove-webworker and run `jlpm link <relative path to cockle directory>`. Then build it in the usual way, you'll need dependencies like `jupyterlite-core` and so on. To serve use
```
jupyter lite serve --LiteBuildConfig.extra_http_headers=Cross-Origin-Embedder-Policy=require-corp --LiteBuildConfig.extra_http_headers=Cross-Origin-Opener-Policy=same-origin
```
When I do this I get
```
Uncaught (in promise) TypeError: __webpack_modules__[moduleId] is not a function
    at __webpack_require__ (bootstrap:19:1)
    at ../cockle/lib/shell_impl.js (history.ts:73:2)
    at __webpack_require__ (bootstrap:19:1)
    at ../cockle/lib/shell_worker.js (shell_impl.ts:420:2)
    at __webpack_require__ (bootstrap:19:1)
    at bootstrap:34:1
    at __webpack_require__.O (chunk loaded:23:1)
    at __webpack_require__.x (bootstrap:35:1)
```
in the browser console.

**Comment by ianthomas23** [comment]
Now I've released `cockle` 0.0.7 I've updated this `remove-webworker` branch to use that, so it is no longer necessary to build `cockle` from source, just building `terminal` from this branch is enough.

**Comment by ianthomas23** [comment]
Closing this as PR #21 is working on a fix.
